### PR TITLE
custom filter checkboxes cms tag created

### PR DESCRIPTION
**Pull request by stanleypliu**
**Ticket**

https://unep-wcmc.codebasehq.com/projects/pp-refresh/tickets/130

Created a custom checkbox component that is designed to be used to populate filters in the News and Stories section. This can be added to the CMS and passed any number of string arguments, as well as whatever name the user wants to add - these arguments are initialised as part of the FilterCheckbox fragment attributes so could probably be accessed using `cms_fragment_content` by the front end or something similiar. Might need a bit of styling, but just want to check that it works as it should first. 

P.S. New branch has been created in the db submodule as I've edited `db/cms_seeds/protected-planet/layouts/template-basic/content.html` to include two of these tags.

### CODE REVIEW

File: app/assets/javascripts/comfy/admin/cms/filters.js.coffee [hunk 1]
```diff
@@ -0,0 +1,14 @@
+$(document).ready( ->
+  # Create a change event listener for each topic checkbox.
+  # This will add or remove the related substring in the content input tag.
+  ['WDPA', 'OECM', 'PAME'].forEach((el) =>
```
Comment (stanleypliu) [inline app/assets/javascripts/comfy/admin/cms/filters.js.coffee:4]: You're using a hard-coded array here of categories, is it possible that we could pass an array to this function instead dynamically?

Comment (stanleypliu) [reply app/assets/javascripts/comfy/admin/cms/filters.js.coffee:4]: But good call with the JavaScript to change the form.

File: app/assets/javascripts/comfy/admin/cms/filters.js.coffee [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    $('#fragment-topic_'+el).change((event) =>
```
Comment (stanleypliu) [inline app/assets/javascripts/comfy/admin/cms/filters.js.coffee:5]: Where does `#fragment-topic` come from? Doesn't seem to be defined in `filter_checkboxes`, so I assume it's somehow being incorporated into the ID of the element.

File: lib/cms_tags/filter_checkboxes.rb [hunk 3]
```diff
@@ -1,57 +1,39 @@
...
-  end
-
-  def store_in_content(category)
-    value = self.send(category)
-    fragment.content << " #{category} #{value} "
+    @custom_categories = options['options'].split(',').map(&:strip)
   end
   
   def form_field(object_name, view, index)
     name = "#{object_name}[fragments_attributes][#{index}][content]"
```
Comment (stanleypliu) [inline lib/cms_tags/filter_checkboxes.rb:8]: Here's what I don't really understand about the variables inside the square brackets - they're not indices of any array, the square brackets seem to mean something else in this context. Although when I changed `boolean` to `content` in the last bracket, I could tell that it's basically specifying which attribute of the current tag to use.

File: app/assets/javascripts/comfy/admin/cms/filters.js.coffee (hunk 1 above)
Comment (stanleypliu) [reply app/assets/javascripts/comfy/admin/cms/filters.js.coffee]: I've changed this now to be dynamic.

File: lib/cms_tags/filter_checkboxes.rb (hunk 3 above)
Comment (Levia) [reply lib/cms_tags/filter_checkboxes.rb:8]: `fragments_attributes` it's not about array indices, it's basically related to the nested object attributes.
For example, here we have a form for a Page object, to assign values to it's nested object, in this case a Fragment, you can use the above syntax

**Comment by stanleypliu** [comment]
Still need to do a couple of things - try to add CSS to the tag dynamically using JavaScript/Coffeescript as it's misaligned at the moment with the label, and also account for the case when there's more than one of these filter checkbox tags present on the page, so there needs to be a way to identify each tag.

**Pull request closed without merging**